lib/worker/batcher.ex

```diff
@@ -528,6 +530,88 @@ defmodule BorsNG.Worker.Batcher do
     :ok
   end
 
+  defp check_code_owner(repo_conn, patch, toml) do
+
+    if !toml.use_codeowners do
+      true
+    else
+
+      Logger.info("Checking code owners")
+      {:ok, code_owner} = Batcher.GetCodeOwners.get(repo_conn, "master")
+      Logger.info("CODEOWNERS file #{inspect(code_owner)}")
+
+
+      {:ok, files} = GitHub.get_pr_files(repo_conn, patch.pr_xref)
+      Logger.info("Files found: #{inspect(files)}")
+
+
+      required_reviews = BorsNG.CodeOwnerParser.list_required_reviews(code_owner, files)
+
+      passed_review = repo_conn
+                      |> GitHub.get_reviews!(patch.pr_xref)
+
+      Logger.info("Passed reviews: #{inspect(passed_review)}")
+
+      # Convert the list of required reviewers into a list of true/false
+      # true indicates that the reviewers requirement was satisfied,
+      # false if it is open
+      approved_reviews = Enum.map(required_reviews, fn x ->
+
+        # Convert a list of OR reviewers into a true or false
+        Enum.reduce(x, false, fn required, acc ->
```

Comment:
`Enum.reduce(x, false, fn _bla, acc -> acc || some_boolish_expr end)` is equivalent to [`Enum.any?(x, fn _bla -> some_boolish_expr end)`](https://hexdocs.pm/elixir/Enum.html#any?/2). In this case, your code can be rewritten to:

```elixir
Enum.any?(x, fn required ->
    if String.contains?(required, "/") do
        # Remove leading @ for team name
        # Split into org name and team name
        team_split = String.slice(required, 1, String.length(required)-1)
                     |> String.split("/")

        # Lookup team ID -> needed later
        {:ok, team} = GitHub.get_team_by_name(repo_conn, Enum.at(team_split, 0), Enum.at(team_split, 1))

        Logger.info("Team: #{inspect(team)}")

        # Loop through reviewers, if they on the team accept their approval
        team_approved = Enum.any?(passed_review["approvers"], fn x ->
          GitHub.is_user_on_team!(repo_conn, x, team.id)
        end)
        Logger.info("Approved: #{inspect(team_approved)}")
        team_approved
    end
end)

Reply:
⚡️